app/services/pdf_parsers/umc_pathology_parser.rb

```diff
@@ -0,0 +1,125 @@
+module PdfParsers
+  class UmcPathologyParser
+    def initialize(pdf_path)
+      @pdf_path = pdf_path
+      @processed_data = {
+        'patient_info' => {},
+        'categories' => {
+          'Basic Metabolic Panel' => [],
+          'Complete Blood Count' => [],
+          'Other Tests' => []
+        }
+      }
+      @current_category = nil
+      @debug_info = []
+      
+      Rails.logger.debug "Initialized UMC Pathology Parser with path: #{pdf_path}"
+    end
+
+    def parse
+      Rails.logger.info "Starting UMC Pathology parse"
+      begin
+        reader = PDF::Reader.new(@pdf_path)
```

Comment:
You can initialize this in constructor as you have already had `@pdf_path` there.